mmdet/datasets/openimages.py

```diff
@@ -0,0 +1,845 @@
+                - `label` (int): the label id.
+                - `is_group_of` (bool):  Indicates that the box spans a group
+                    of objects (e.g., a bed of flowers or a crowd of people).
+                - `is_occluded` (bool): Indicates that the object is occluded
+                    by another object in the image.
+                - `is_truncated` (bool): Indicates that the object extends
+                    beyond the boundary of the image.
+                - `is_depiction` (bool): Indicates that the object is a
+                    depiction.
+                - `is_inside` (bool): Indicates a picture taken from the
+                    inside of the object.
+            `data_infos` (list[dict])： Data infos where each item of
+            the list indicates an image. Keys of annotations are:
+
+                - `img_id` (str): Image name.
+                - `filename` (str): Image name with suffix.
+        """
+
+        item_list = defaultdict(list)
+        data_infos = []
+        cp_filename = None
+        with open(ann_file, 'r') as f:
+            reader = csv.reader(f)
+            i = -1
+            for line in reader:
+                i += 1
+                if i == 0:
+                    continue
+                else:
+                    img_id = line[0]
+                    filename = f'{img_id}.jpg'
+                    label_id = line[2]
+                    assert label_id in self.index_dict
+                    label = int(self.index_dict[label_id])
+                    bbox = [
+                        float(line[4]),  # xmin
+                        float(line[6]),  # ymin
+                        float(line[5]),  # xmax
+                        float(line[7])  # ymax
+                    ]
+                    is_occluded = True if int(line[8]) == 1 else False
+                    is_truncated = True if int(line[9]) == 1 else False
+                    is_group_of = True if int(line[10]) == 1 else False
+                    is_depiction = True if int(line[11]) == 1 else False
+                    is_inside = True if int(line[12]) == 1 else False
+
+                    item_list[img_id].append(
+                        dict(
+                            bbox=bbox,
+                            label=label,
+                            is_occluded=is_occluded,
+                            is_truncated=is_truncated,
+                            is_group_of=is_group_of,
+                            is_depiction=is_depiction,
+                            is_inside=is_inside))
+                    if filename != cp_filename:
+                        data_infos.append(
+                            dict(img_id=img_id, filename=filename))
+                        cp_filename = filename
+        return item_list, data_infos
+
+    def load_annotations(self, ann_file):
+        """Load annotation from annotation file."""
+        self.ann_infos, data_infos = self.list_from_csv(ann_file)
```

Comment:
We should put the logic in self.list_from_csv directly in load_annotations to avoid unnecessary encapsulation.